app/views/project/editProject.js

```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
       })
 }
 
+let DeleteProjectDialog = component('DeleteProjectDialog', (cursor) => {
+  
+  let yesCallback = () => {
+    let editCursor = cursor.cursor('editProject')
+    let projectCursor = editCursor.cursor('project')
+    let project = projectCursor.toJS()
+    let qualifiedProjectId = `${project.owner}/${project.projectId}`
+    
+    editCursor.set('isWaiting', true)
+    
+    ajax.delete(`/api/projects/${project.owner}/${project.projectId}`)
+      .then(() => {
+        logger.debug(`Project successfully deleted '${qualifiedProjectId}'`)
+        router.goTo('/')
+      }, (error) => {
+        logger.warn(`Failed to delete project '${qualifiedProjectId}': ${error}`)
+      })
+      .finally(() => {
+        cursor.set('showDeleteProjectDialog', false)
+        editCursor.set('isWaiting', false)
+      })
+  }
+
+  let closeCallback = () => {
+    cursor.set('showDeleteProjectDialog', false)
+  }
+  let title = 'Delete'
+  let message = 'Are you sure you want to delete this project?'
+
+  return notification.question(title, message, yesCallback, closeCallback)
+  
```

Comment:
Just a personal opinion about readability and breathing room. A lot of the source code has very little breathing room, and I guess I just felt compelled to add more space. I added a newline on the top of the function too.

Take the code below for example:
```javascript
let editProject = (cursor) => {
  let editCursor = cursor.cursor('editProject')
  let editProject = editCursor.toJS()
  uploadProject(editProject.project, editCursor, cursor)
    .then(({title, description, instructions, tags, licenseId, username, pictureFiles, files,}) => {
      logger.debug(`Picture files:`, pictureFiles)
      logger.debug(`Files:`, files)
      logger.debug(`title: ${title}, description: ${description}, tags: ${S.join(`,`, tags)}`)
      let qualifiedProjectId = `${editProject.owner}/${editProject.projectId}`
      let data = {
        title,
        description,
        instructions,
        tags,
        licenseId,
        pictures: pictureFiles,
        files,
      }
      logger.debug(`Updating project '${qualifiedProjectId}'...:`, data)
      ajax.putJson(`/api/projects/${qualifiedProjectId}`, data)
```